x/distribution/keeper/allocation.go

```diff
@@ -78,6 +78,11 @@ func (k Keeper) AllocateTokens(ctx context.Context, totalPreviousPower int64, bo
 		return err
 	}
 
+	// set ToDistribute in protocolpool to keep track of continuous funds distribution
+	if err := k.poolKeeper.SetToDistribute(ctx, amt); err != nil {
```

Comment:
Is there another way to keep track of it? I would rename the keeper call to something more clear.

Reply:
protocolpool doesnt know how much was sent to the community pool. If its set in pool module, it would always assume its the entire pool unless you started tracking the community pool amount somewhere. So, marko suggested that distribution should set toDistibute in allocate tokens with how much was sent there.
I'm not sure if there are any other ways to do so.

Reply:
Protocolpool could take a snapshot of its balance at every block.
Maybe it is expensiver, but it would keep the logic within protocolpool

Reply:
any thoughts on this @tac0turtle

Reply:
so there are two things to consider, if a user sends funds to the community pool, is it to the community pool or to all the funds and streams? If its the latter then we cant do a snapshot every block, if its the former then we need to adjust the logic to handle multiple denominations. 

this setting makes an explicit dependency between the two modules, the snapshot does an implicit. The goal is to make things lazy in this pr. Protocolpool doesnt need anything in begin or end block imo

Reply:
That is basically my issue. When we drop legacy support of distribution community pool spend message, then distribution does not need to import protocolpool. Currently, the dependency is already there for backward compatibility version, but hopefully for v0.52 we can remove it, when everyone has updated to use directly the protocolpool messages.
The dependency is already implicit, as it is based on sending funds to a module account, and those are defined in auth.
Imho `x/protocolpool` shouldn't need to be imported by any module and be standalone, which does involve begin/end block indeed. 🤷🏾‍♂️

Reply:
Concerning the fact that funding the community pool, would be taxed and fund the funds and streams as well, that is already for streams, so why not for funds?

Reply:
this isnt introducing a dependency import, this is only an expected interface. I dont quite follow the import issue? 

Begin block and endblock create a complex dependency graph. This is creating an explicit dependency graph, not through imports. In the readme we will say it is required to send funds to the account and call this function. In the future with post message hooks we can get rid of this dependency since we will say anytime someone sends funds to the protocol pool account we will add to `toDistribute`, but for now this is the easiest approach .  

streams and funds have a different design. Streams take from the community pool, while funds go before the funds are part of the community pool. This avoids needing to integrate funds into distribution.

Reply:
> this isnt introducing a dependency import, this is only an expected interface. I dont quite follow the import issue?

You are right about that. I have missed that we don't accept the type directly.

> streams and funds have a different design. Streams take from the community pool, while funds go before the funds are part of the community pool. This avoids needing to integrate funds into distribution.

Not sure what you mean by "funds go before the funds are part of the community pool".
Distribution sends them all to `x/protocolpool`. Then, when you withdraw it, withdraw from the community pool.
Those funds are currently not blocked, and you can actually use them until withdrawn. A community pool spends proposal can pass and empty the pool, before some funds actively withdraw their funds -- that's the drawback of having it lazy. Probably a bug in the current implementation, those funds should maybe be moved to another module account.

Reply:
@likhita-809 can you confirm the behavior with what Marko said?